feat: support disabling remote log uploads

mnahkies opened the pull request:
relates: https://github.com/tailscale/tailscale/issues/13174

- adds a new switch to the settings page for enabling/disabling remote log uploads

- calls the `Disable` function from the `logtail` package during init when the setting is turned off

ref: https://pkg.go.dev/tailscale.com/logtail#Disable

<details><summary>Expand for Screenshot</summary>
<p>

<img width="728" height="913" alt="image" src="https://github.com/user-attachments/assets/ea224459-2860-48cc-968b-6b6ce305ffd0" />

</p>
</details>

mnahkies committed:
feat: support disabling remote log uploads

Updates https://github.com/tailscale/tailscale/issues/13174

- adds a new switch to the settings page for
  enabling/disabling remote log uploads

- calls the `Disable` function from the `logtail`
  package during init when the setting is turn off

ref: https://pkg.go.dev/tailscale.com/logtail#Disable

Signed-off-by: Michael Nahkies <michael@nahkies.co.nz>

bradfitz commented on android/src/main/res/values/strings.xml:
```diff
@@ -346,6 +346,8 @@
     <string name="use_tailscale_subnets_subtitle">Route traffic according to your network\'s rules. Some networks require this to access IP addresses that don\'t start with 100.x.y.z.</string>
     <string name="subnet_routing">Subnet routing</string>
+    <string name="client_remote_logging_enabled">Remote client logging</string>
+    <string name="client_remote_logging_enabled_subtitle">Equivalent to --no-logs-no-support on Linux.\nChanges require restarting the app to take effect.</string>
```
This is pretty nerdy. We don't need to reference the flag name or Linux here and can just explain the implications:

No debug logs, no support, prevents using Network Flow logs, etc. 

Like "Whether debug logs are uploaded to Tailscale support."

It's admittedly hard to fit all the nuance in a couple lines of text.

mnahkies replied:
It's a bit wordy, but perhaps something like this?
<img width="679" height="168" alt="image" src="https://github.com/user-attachments/assets/89dd8ecd-d82e-4fcc-b919-9cfdc37c6e8f" />

Couple other thoughts:
- Should the "Bug report" button be hidden when logs are disabled?
- Is this fine as a top-level settings item, or would it be better under "Permissions"?
- How does / should this interact with MDM settings? (I fear I might be opening a can of worms with this one, as it might require backend changes to closed source components? :sweat_smile:)

mnahkies replied:
Another thought on MDM: assuming I'm correct that this would need to be controllable through MDM (to enforce flow logs are produced for enterprise) - is there a stepping stone where we could force this setting on when any MDM is employed, and then later make it an individually configurable MDM setting?

mnahkies replied:
I've taken a stab at implementing what I think respecting MDM _could_ look like in https://github.com/tailscale/tailscale-android/pull/695/commits/04fd66c55f7e29b10ebab1b64890adf7356cfb2e (has an outstanding todo to handle async config updates correctly, and I haven't been able to enroll a device to test e2e properly yet)

TBH for my purposes, I can probably just run a custom build with this patch applied, though I do think it would be nice to land in `main` - particularly in the headscale use-case, sending logs to tailscale doesn't make a lot of sense to me, which as far as I can tell would currently occur.

mnahkies committed:
fix: tweak copy

Signed-off-by: Michael Nahkies <michael@nahkies.co.nz>

mnahkies committed:
fix: force client logging on when any mdm is configured

Signed-off-by: Michael Nahkies <michael@nahkies.co.nz>

generic-internet-user commented:
any progress on this?

fartoonice commented:
Eagerly awaiting this implementation so I can finally start using what I otherwise deem to be a pretty cool project.

Mynacol commented:
Hey, we are again two months in without any reaction. Can someone from Tailscale bring this forward? @Erisa maybe?

generic-internet-user commented:
I'd be interested in seeing an official response to this as well by now

bradfitz committed:
android: require confirmation before disabling remote client logging

Reword the subtitle and add a confirmation dialog when toggling
remote client logging off, since disabling breaks Network Flow
Logs (if required by the tailnet admin) and blocks Tailscale
Support from debugging problems on the device. Turning logging
back on remains a single tap.

Signed-off-by: Brad Fitzpatrick <bradfitz@tailscale.com>

bradfitz commented:
Looks good to us.

I rebased this locally and fixed some of the English up and it looks like the MDM check is backwards, so fixing + verifying that too. But first I sent out https://github.com/tailscale/tailscale/pull/19446 to remove the "you must restart the process" thing which is awkward.

I'll update this PR and merge it once that other change is in and I've verified it works.

bradfitz committed:
libtailscale,android: toggle remote client logging without restarting

Switch the remote-logging toggle from the one-way process-wide
logtail.Disable kill switch to per-logger APIs added upstream:
Logger.SetEnabled for runtime flips, and Config.Disabled so the
first NewLogger call doesn't even emit the internal "logtail
started" banner when the user has opted out. Bumps go.mod to
tailscale.com 1e68a11721fd which includes both.

The setting now takes effect immediately with no restart required.
App.updateIsClientLoggingEnabled pushes the effective value (still
forced on while MDM is configured) to the backend, and the MDM
change receiver does the same on MDM transitions so an arriving
MDM profile forces logging back on live.

Also fix the inverted MDMSettings.isMDMConfigured check so it tracks
"MDM restrictions are set" rather than its inverse; without this the
prior commit silently did the opposite of what it claimed. Drop the
"changes require restarting the app to take effect" wording from the
settings subtitle and disable-confirmation dialog.

Updates #13174

Signed-off-by: Brad Fitzpatrick <bradfitz@tailscale.com>
